src/validation.cpp

```diff
@@ -2526,8 +2512,15 @@ bool ActivateBestChain(CValidationState &state, const CChainParams& chainparams,
             fInitialDownload = IsInitialBlockDownload();
 
             // throw all transactions though the signal-interface
-
-            } // MemPoolConflictRemovalTracker destroyed and conflict evictions are notified
+            connectTrace.CallSyncTransactionOnConflictedTransactions();
```

Comment:
Argh, I believe that could be a race which triggers a reachable assertion if there are two ActivateBestChain's going at once, moved the class into the cs_main lock, which simplified the whole thing a bit.